homeassistant/components/ondilo_ico/__init__.py

```diff
@@ -0,0 +1,59 @@
+"""The Ondilo ICO integration."""
+import asyncio
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import config_entry_oauth2_flow
+
+from . import api, config_flow
+from .const import DOMAIN
+from .oauth_impl import OndiloOauth2Implementation
+
+PLATFORMS = ["sensor"]
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the Ondilo ICO component."""
+    hass.data.setdefault(DOMAIN, {})
+
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Ondilo ICO from a config entry."""
+
+    config_flow.OAuth2FlowHandler.async_register_implementation(
```

Comment:
As far as I can tell, when using the config flow, async_setup or async_setup_entry are not called before the call to async_step_user, so I need that call in async_step_user. Then, when restarting hass, I also need to register the OAuthImplementation so I need to make sure it's declared either in async_setup or in async_setup_entry. As I don't support the yaml declaration, I chose to put that call in async_setup_entry. Defining it only in async_setup is not enough.
Am I wrong here?

Reply:
Yeah, that makes sense.  I see what I was missing.  I got this from https://github.com/home-assistant/core/blob/dev/script/scaffold/templates/config_flow_oauth2/integration/__init__.py#L42  The difference may just be that in that example a `configuration.yaml` is needed for configuring oauth client id and client secret, and that would invoke `async_setup`. Not the case here since its hard coded.

Reply:
Do you know if you actually need to call `async_register_implementation` or `async_get_config_entry_implentation` here? It may be sufficient to just pass `OndiloOauth2Implementaiton` into `OndiloClient` since the config flow will register itself.